styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorageTransaction.java

```diff
@@ -175,4 +188,36 @@ public WorkflowInstance deleteActiveState(WorkflowInstance instance) {
     tx.delete(activeWorkflowInstanceKey(tx.getDatastore().newKeyFactory(), instance));
     return instance;
   }
+
+  @Override
+  public Backfill store(Backfill backfill) {
+    final Key key = DatastoreStorage.backfillKey(tx.getDatastore().newKeyFactory(), backfill.id());
+    Entity.Builder builder = Entity.newBuilder(key)
+        .set(PROPERTY_CONCURRENCY, backfill.concurrency())
+        .set(PROPERTY_START, instantToTimestamp(backfill.start()))
+        .set(PROPERTY_END, instantToTimestamp(backfill.end()))
+        .set(PROPERTY_COMPONENT, backfill.workflowId().componentId())
+        .set(PROPERTY_WORKFLOW, backfill.workflowId().id())
+        .set(PROPERTY_SCHEDULE, backfill.schedule().toString())
+        .set(PROPERTY_NEXT_TRIGGER, instantToTimestamp(backfill.nextTrigger()))
+        .set(PROPERTY_ALL_TRIGGERED, backfill.allTriggered())
+        .set(PROPERTY_HALTED, backfill.halted());
+
+    backfill.description().ifPresent(x -> builder.set(PROPERTY_DESCRIPTION, StringValue
+        .newBuilder(x).setExcludeFromIndexes(true).build()));
+
+    tx.put(builder.build());
+
+    return backfill;
+  }
+
+  @Override
+  public Optional<Backfill> backfill(String id) {
+    final Key key = DatastoreStorage.backfillKey(tx.getDatastore().newKeyFactory(), id);
+    final Entity entity = tx.get(key);
+    if (entity == null) {
+      return Optional.empty();
+    }
+    return Optional.of(entityToBackfill(entity));
```

Comment:
Minor cleanup for later: why backfillToEntity was ported to here but entityToBackfill wasn't?

Reply:
Good point, we'll need a proper refactoring for these Datastore classes in future PR.